### PR TITLE
Yejiaojiao/dev bcewithlogitsloss

### PR DESCRIPTION
**Pull request by JiaojiaoYe1994**
add BCEWithLogitLoss API in Eager mode for Of 0.3.4 module.

* docstring:
bcewithlogitsloss_docstring

* doctest:
bcewithlogitsloss_doctest

* test case:
bcewithlogitsloss_testcase

### CODE REVIEW

File: docs/source/experimental.rst [hunk 1]
```diff
@@ -128,6 +128,8 @@ Experimental features
 .. autofunction:: oneflow.experimental.Tensor.pow
 .. autofunction:: oneflow.experimental.cosh
 .. autofunction:: oneflow.experimental.Tensor.cosh
+.. autofunction:: oneflow.experimental.acos
+.. autofunction:: oneflow.experimental.Tensor.acos
```
Comment (BBuf) [inline docs/source/experimental.rst:132]: 你这个module是bcewithlogitloss，为什么把acos也放到这个pr里面？

File: oneflow/python/nn/modules/loss.py [hunk 2]
```diff
@@ -296,7 +296,151 @@ def forward(self, input, target):
...
+
+        out =k -weight*\sum_{i=1}^n[Pos\_weight*y*log\sigma({x}) + (1-y)*log(1-\sigma(x))]
+
+    Args:
+        input (oneflow._oneflow_internal.BlobDesc): The input Tensor.
+        target (oneflow._oneflow_internal.BlobDesc): The target Tensor.
+        weight (remote_blob_util, optional): The manual rescaling weight to the loss. Defaults to None.
+        pos_weight (remote_blob_util, optional): The manual rescaling weight to the positive examples. Defaults to None.
+        reduction (str, optional): The reduce type, it can be one of "none", "mean", "sum". Defaults to "mean".
+        name (Optional[str], optional): The name for the operation. Defaults to None.
```
Comment (BBuf) [inline oneflow/python/nn/modules/loss.py:332]: name不需要了，可以删除

Comment (JiaojiaoYe1994) [reply oneflow/python/nn/modules/loss.py:332]: 好的，已删除

File: oneflow/python/test/modules/test_bcewithlogitsloss.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    else:
+        return _weighted_loss
+
+
+@unittest.skipIf(
+    not flow.unittest.env.eager_execution_enabled(),
+    ".numpy() doesn't work in lazy mode",
+)
+class TestBCEWithLogitsLossModule(flow.unittest.TestCase):
+    def test_bcewithlogitsloss_none(test_case):
```
Comment (BBuf) [inline oneflow/python/test/modules/test_bcewithlogitsloss.py:72]: 单测需要重构，参考其它pr。

Comment (JiaojiaoYe1994) [reply oneflow/python/test/modules/test_bcewithlogitsloss.py:72]: 查看了下各loss的测试方法不同，参考https://github.com/Oneflow-Inc/oneflow/pull/5116/files `MSELoss` 这个方式重构？

File: oneflow/python/nn/modules/loss.py [hunk 4]
```diff
@@ -605,6 +605,148 @@ def forward(self, input1, input2, target):
...
+
+    def __init__(
+        self,
+        weight = None,
+        size_average: bool = True,
+        reduce: bool = True,
+        reduction: Optional[str] = "mean",
+        pos_weight = None,
+    ) -> None:
+        ], "only 'sum', 'mean' and None supported by now"
```
Comment (BBuf) [inline oneflow/python/nn/modules/loss.py:691]: 这一行是打错了？

Comment (BBuf) [reply oneflow/python/nn/modules/loss.py:691]: 你这程序能编译过吗

File: oneflow/python/test/modules/test_crossentropyloss.py [hunk 5]
```diff
@@ -13,14 +13,10 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 """
+import oneflow.experimental as flow
```
Comment (BBuf) [inline oneflow/python/test/modules/test_crossentropyloss.py:16]: 为什么要改这个文件呢？这是另外一个module的测试。

File: oneflow/python/test/modules/test_nllloss.py [hunk 6]
```diff
@@ -14,12 +14,9 @@
 limitations under the License.
 """
 import unittest
```
Comment (BBuf) [inline oneflow/python/test/modules/test_nllloss.py:16]: 同理，这个文件也不应该被改动。

File: oneflow/python/nn/modules/loss.py (hunk 4 above)
Comment (JiaojiaoYe1994) [reply oneflow/python/nn/modules/loss.py:691]: 失误， 我修改下 ;/

**Review by BBuf — APPROVED** [review]

File: oneflow/python/nn/modules/loss.py [hunk 7]
```diff
@@ -606,6 +606,154 @@ def forward(self, input1, input2, target):
...
+        out = -\frac{weight}{n}\sum_{i=1}^n[Pos\_weight*y*log\sigma({x}) + (1-y)*log(1-\sigma(x))]
+
+    if :attr:`reduction` = ``"sum"``:
+
+    .. math::
+
+        out =k -weight*\sum_{i=1}^n[Pos\_weight*y*log\sigma({x}) + (1-y)*log(1-\sigma(x))]
+
+    Args:
+        weight (remote_blob_util, optional): The manual rescaling weight to the loss. Default: ``None``
```
Comment (Flowingsun007) [inline oneflow/python/nn/modules/loss.py:636]: 这个是weight是Tensor，remote_blob_util去掉吧

File: oneflow/python/nn/modules/loss.py [hunk 8]
```diff
@@ -606,6 +606,154 @@ def forward(self, input1, input2, target):
...
+    def forward(self, input, target):
+        if not (target.shape == input.shape):
+            raise ValueError("Target size ({}) must be the same as input size ({})".format(target.size(), input.size()))
+
+        _neg_input = flow.experimental.negative(input)
+        _max_val = flow.experimental.clip(_neg_input,0)
+        _neg_max_val = flow.experimental.negative(_max_val)
+
+        if self.pos_weight:
+            assert self.pos_weight.shape[0] == input.shape[-1], (
```
Comment (Flowingsun007) [inline oneflow/python/nn/modules/loss.py:725]: 这块会不会有问题（能确保n_class所在维度都是input shape的-1维吗？

File: oneflow/python/nn/modules/loss.py [hunk 9]
```diff
@@ -606,6 +606,160 @@ def forward(self, input1, input2, target):
...
+            None,
+        ], "only 'sum', 'mean' and None supported by now"
+
+        self.weight = weight
+        self.size_average = size_average
+        self.reduce = reduce
+        self.reduction = reduction
+        self.pos_weight = pos_weight
+        self._transpose_op = (
+            flow.builtin_op("transpose")
```
Comment (Flowingsun007) [inline oneflow/python/nn/modules/loss.py:705]: 建议去掉`flow.builtin_op("transpose")`,可以直接在forward里用flow.experimental.transpose或permute，后面也方便重构成functional的方式调用

File: oneflow/python/nn/modules/loss.py (hunk 7 above)
Comment (JiaojiaoYe1994) [reply oneflow/python/nn/modules/loss.py]: 已修改。

File: oneflow/python/nn/modules/loss.py (hunk 9 above)
Comment (JiaojiaoYe1994) [reply oneflow/python/nn/modules/loss.py:705]: 好的，已根据建议进行了修改。

File: oneflow/python/nn/modules/loss.py (hunk 8 above)
Comment (JiaojiaoYe1994) [reply oneflow/python/nn/modules/loss.py:725]: 查看了下pytorch的实现，在后面的矩阵操作会自动比对维度，所以这部分没必要我给删去了。
